lib/Parse/Lexer.cpp

```diff
@@ -1645,13 +1707,14 @@ static void validateMultilineIndents(const Token &Str,
 /// lexStringLiteral:
 ///   string_literal ::= ["]([^"\\\n\r]|character_escape)*["]
 ///   string_literal ::= ["]["]["].*["]["]["] - approximately
-void Lexer::lexStringLiteral() {
+void Lexer::lexStringLiteral(unsigned DelimiterLength) {
   const char *TokStart = CurPtr-1;
   assert((*TokStart == '"' || *TokStart == '\'') && "Unexpected start");
```

Comment:
Please adjust `TokStart` with `DelimiterLength` as soon as possible. Otherwise, for unterminated string such as `#"foo<EOF>`, `#` is not included in the unknown token.

Reply:
Difficult to know what to do without broader changes to lexStringLiteral(). This is fixed though.